### DISCUSSION TITLE
Option to swap README position with Repository Code

### DISCUSSION THREAD

**Discussion by xetra11** [community/community#3048] [Other Feature Feedback, Questions, & Ideas]
I figured out that the first thing I do when visiting a repository is to scroll down to the Readme. Therefore I wonder why the Readme is not the first thing to see of a repository. I kinda feel that seeing the code first does not really help unless the repository only contains a script snippet file or something else.

What I'd like to see I reproduced in a single javascript snippet:

```javascript
// author: xetra11
// https://github.com/xetra11

{   
    function htmlToElement(html) {
        var template = document.createElement('template');
        html = html.trim();
        template.innerHTML = html;
        return template.content.firstChild;
    }

    const swap = function(nodeA, nodeB) {
        const parentA = nodeA.parentNode;
        const siblingA = nodeA.nextSibling === nodeB ? nodeA : nodeA.nextSibling;

        // Move `nodeA` to before the `nodeB`
        nodeB.parentNode.insertBefore(nodeA, nodeB);

        // Move `nodeB` to before the sibling of `nodeA`
        parentA.insertBefore(nodeB, siblingA);
    };

    const neighborButtonNode = document.body.querySelector(".btn.ml-2.d-none.d-md-block")
    const swapButtonNode = htmlToElement('<a class="btn ml-2 d-none d-md-block swap-button" href>Swap View</a>')

    swapButtonNode.addEventListener("click", (event) => { 
        event.preventDefault()
        event.stopPropagation()
        const readmeNode = document.body.querySelector("#readme")
        const codeNode = document.body.querySelector(".Box.mb-3")
        swap(readmeNode, codeNode)
    })

    neighborButtonNode.parentNode.insertBefore(swapButtonNode, neighborButtonNode)
}
```
(works on Chrome and for the today's date. If GitHub has released some changes to the DOM this script might not work anymore)

Run it in your console and see a swap button appear at the top of your visited repository.

![image](https://user-images.githubusercontent.com/8149023/102752146-66304a80-4369-11eb-9260-04d56c7e2256.png)

I'd love to have that feature like the dark mode so I can decide whether I'd like see the readme or the code first!

:wave:
Charlie

**Comment by X39** [comment]
Instead of the button, what about keeping the position as it is right now and instead making the code section expand/retract once clicking somewhere in the "header section" of the code stuff?

there are other areas in github where that already works, so it would be nice if that addition could be added here too
github might even remember my last choice, making it less of a hazzle to actually use if someone always wants the code to be open

**Reply by xetra11** [reply]
Automagically moving the sections would be nice. However imho the Readme should receive way more attention nowadays. So I'd prefer having the Code Section pop up on demand instead of appear priotized by default. I see too few use cases for having to browse the code of a repo. But that's just my opinion since I prefer cloning a repo and browse the code via my IDE instead of clicking through the Github Code View
